xcodeproj/code_sign_info.go

```diff
@@ -146,80 +149,71 @@ func firstNonEmpty(values ...string) string {
 
 // ResolveCodeSignInfo ...
 func ResolveCodeSignInfo(projectOrWorkspacePth, scheme, configuration, user string) (map[string]CodeSignInfo, error) {
-	targetCodeSignInfoMap, err := getCodeSignInfoWithXcodeprojGem(projectOrWorkspacePth, scheme, configuration, user)
+	projectTargetsMapping, err := readSchemeTargetMapping(projectOrWorkspacePth, scheme, configuration, user)
 	if err != nil {
 		return nil, err
 	}
 
 	resolvedCodeSignInfoMap := map[string]CodeSignInfo{}
-	for target, codeSignInfo := range targetCodeSignInfoMap {
-		if target == "" {
-			return nil, errors.New("target name is empty")
-		}
 
-		projectPth := codeSignInfo.ProjectPth
-		if projectPth == "" {
-			return nil, fmt.Errorf("failed to resolve which project contains target: %s", target)
-		}
+	if configuration == "" {
+		configuration = projectTargetsMapping.Configuration
+	}
 
-		configuration = codeSignInfo.Configuration
+	for projectPth, targets := range projectTargetsMapping.Targets {
+		for _, targetName := range targets {
+			if targetName == "" {
+				return nil, errors.New("target name is empty")
```

Comment:
print more info in the error: configuration, or "%+v" out the projectTargetsMapping model